apps/accounts/views.py

```diff
@@ -464,3 +465,27 @@ class RestrictedPage(APIView):
 
     def get(self, request, format=None):
         return Response({"msg": "I am a restricted page"}, status=status.HTTP_200_OK)
+
+
+class Moderator(BasePermission):
+    """
+    This class contains everything related to operations
+    belong to Moderator. Moderator user is different from
+    Admin user, but has some level of responsibilities and
+    access to resources.
+    """
+
+    def has_permission(self, request, *args):
+        """Method to check if the given user_id belongs to moderator or not"""
```

Comment:
this code can be optimised to this,

```
    def has_permission(self, request, *args):
        """Method to check if the given user_id belongs to a moderator or not"""

        user_model = get_user_model()

        # check if user_id is valid or contains an improper value
        if not request.user_id or not is_valid_uuid(request.user_id):
            return False

        # check if the given user_id is present or not, if present then check if it's a moderator
        try:
            user = user_model.objects.get(id=request.user_id)
            return user.is_moderator
        except user_model.DoesNotExist:
            return False
```

Imported get_user_model() to dynamically fetch the user model. This is a more flexible approach as it adapts to any custom user model.